### DISCUSSION TITLE
It's getting detected by cloudflare.

### DISCUSSION THREAD

**Discussion by vndhote** [ultrafunkamsterdam/undetected-chromedriver#1387] [General]
undetected chromedriver worked well till yesterday but now, cloudflare improved and the chromedriver is not bypassing cloudflare. I have attached the screenshot of it. cloudflare is just looping the captcha when selenium is running. When I close it, the website loads
![image](https://github.com/ultrafunkamsterdam/undetected-chromedriver/assets/79296613/1f732ebb-59eb-45a7-8786-86ae6082af44)

**Comment by ParhamSobhan** [comment]
I have the exact same problem, this issue started like 90 minutes ago is there any fix for this?

**Comment by vahner** [comment]
Yes, I also

**Comment by Tssukeiro** [comment]
I have the same issue as you guys, last time i used chromedriver undetected it worked fine (4 days ago)
But today website using cloudflare protection is not bypassed.

**Reply by ysfmrz** [reply]
me too

**Reply by NickxDxx** [reply]
Finally, this project suck as hell! :)

**Comment by vndhote** [comment]
guys I guess we need to wait until new logics are added to undetected chrome driver to make it more stealthy and undetectable.

**Reply by Tssukeiro** [reply]
maybe a different browser is not detected like chrome ? any ideas ?

**Comment by marcogaruthix** [comment]
same here

**Comment by maroun2** [comment]
I have the same problem tested on multiple IPs.

**Comment by damodharheadrun** [comment]
Same here as well

**Comment by vahner** [comment]
new version, which will be able to bypass this, in the near future is not expected?

**Reply by foreandr** [reply]
It worked for about a day and then it broke again

**Comment by damodharheadrun** [comment]
Guys, Please follow the below ticket
https://github.com/ultrafunkamsterdam/undetected-chromedriver/issues/1398

you will get know about cloudflare dedection and bypass the security

**Reply by ghost** [reply]
not working anymore

**Reply by damodharheadrun** [reply]
which method have you tried and can you share the steps I will guide you to help avoid getting detected from the Cloudflare

**Reply by xoxma1993** [reply]
import undetected_chromedriver as uc
import time
chromedriver_path = 'chromedriver'  # Replace 'chromedriver' with the actual path
options = uc.ChromeOptions()
options.add_argument('--headless')
options.add_argument('--no-sandbox')  # Add this argument to avoid sandbox issues on Linux
options.add_argument("--auto-open-devtools-for-tabs") 
driver = uc.Chrome(executable_path=chromedriver_path, options=options)
driver.get('https://nowsecure.nl')
time.sleep(5)
driver.save_screenshot('nowsecure.png')
driver.quit()

**Reply by rcarvalho16** [reply]
Only works because selenium is not interacting with the webpage...

**Reply by developsit** [reply]
if you go to the page add a tab open the devtool go through the cloud return to the tab then selenium should continue to interact and the window with cloudflare should not be disturbed. I'll test at home

**Reply by developsit** [reply]
Another option is to use a new tab using javascript

**Reply by rcarvalho16** [reply]
Have you tried it? If so, how did it go?

**Reply by listmen001** [reply]
It's work!!!
Thanks< from Russian with love!)

**Reply by developsit** [reply]
> It's work!!!
> 
> Thanks< from Russian with love!)

I am not russian. I hate russians! I'm from Ukraine

**Reply by addame2** [reply]
> It's work!!! Thanks< from Russian with love!)

bro do you have telegram: please knock me thanks @iamdev2

**Comment by jctw2020** [comment]
opening dev tools is not working anymore

**Reply by damodharheadrun** [reply]
Can we connect once, so I can help you?  on that

**Comment by myalcin81** [comment]
It's so weird. Opening a new tab works when you open the first tab by keyboard shortcut :) (no repro every time)
here is the log they caught and their obfuscated JS file for analysis, I couldn't fully reverse it it takes time:
[1337.txt](https://github.com/ultrafunkamsterdam/undetected-chromedriver/files/12147548/1337.txt)
You can see undedected chromedriver 1337 console log in devtool :)

**Comment by ultrafunkamsterdam** [comment]
just use :
```
with driver:
    driver.get('https://somecloudflareprotectedsite.tld')
    
```
tested and worked. when i connect to nordvpn it stops working

**Reply by ysfmrz** [reply]
I also tested this method. But it didn't work
Cloudflare is repeated in a loop.

Should we wait for the new update? for solving this problem?
